tests/Serializer/JsonSerializationTest.php

```diff
@@ -423,6 +425,18 @@ public static function getTypeHintedArraysAndStdClass()
         ];
     }
 
+    public function testDeserializingUnionProperties()
```

Comment:
this seems doing the test with primitives... what about union of objects? is that supported?